lib/rules/avoid-leaking-callbacks-in-ember-objects.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+// const ember = require('../utils/ember');
+const utils = require('../utils/utils');
+
+const ERROR_MESSAGE = 'All `addEventListener` calls should have a corresponding `removeEventListener` to avoid leaking callbacks.';
+//------------------------------------------------------------------------------
+// Ember object rule - Avoid callback memory leaks
```

Comment:
Should this rule be limited to Ember objects only?

If it should, it would need a `if (!(ember.isEmberObject(node) || ember.isReopenObject(node)))` check somewhere, similar to the existing `avoid-leaking-state-in-ember-objects` rule.

If it's not necessary to limit it to Ember objects, and we can just run it on a per-file basis, then we would want to rename the rule to avoid mentioning `in-ember-objects`, and remove all the extra Ember object code from the test cases.

Reply:
I think we can limit only for Ember objects for now, because for other objects, this will be a little out of scope for the plugin itself. Will make the changes accordingly

Reply:
@bmish I tend to agree. I don't see a reason to scope this to just Ember objects.

Reply:
@rajasegar I wonder if you could try contributing this rule to eslint itself as a general `no-leaking-callbacks` rule? They are pretty strict about accepting new rules but perhaps you could at least ask what they think about it since it seems like a general JavaScript rule.

Reply:
Sure @bmish I will try